### PR TITLE
fix(gradle): Stop return non-version "currentValue" fields

### LINKED ISSUE #7300: The pull-request description contains packages that was never updated
**What Renovate type, platform and version are you using?**

Self-hosted Renovate, version 23.19.4
Bitbucket Server, version 7.1.1

**Describe the bug**

The pull-request description generated by Renovate contains packages that was never updated. This will confuse the pull-request reviewers.

The extra packages that shouldn't be there are those packages in the build.gradle file that has no specified version.

**Relevant debug logs**

I cannot find any errors in the log, but I can upload a complete log if needed.

**To Reproduce**

This repository will result in a pull-request containing packages that was never updated:
https://github.com/niclastegner/example-gradle-with-renovate

A screenshot of the pull-request description in Bitbucket showing that the first package in the list is correct, but all the `org.springframework.boot:* `shouldn't be there:

[pull-request-description.png](https://github.com/niclastegner/example-gradle-with-renovate/blob/master/screenshots/pull-request-description.png)

A screenshot of the diff i Bitbucket:

[pull-request-diff.png](https://github.com/niclastegner/example-gradle-with-renovate/blob/master/screenshots/pull-request-diff.png)

**Additional context**

None

### PR DESCRIPTION
**Pull request by zharinov**
Ref #6477 
Closes #7300
Example: https://github.com/zharinov/example-gradle-with-renovate/pull/1

### CODE REVIEW

File: lib/manager/gradle/build-gradle.ts [hunk 1]
```diff
@@ -212,7 +212,12 @@ export function collectVersionVariables(
     }
 
     if (!dep.currentValue && variables[depName]) {
-      dep.currentValue = variables[depName];
+      const isVersionLiteral = !/^[a-zA-Z_][a-zA-Z0-9_.]*$/.test(
```
Comment (viceice) [inline lib/manager/gradle/build-gradle.ts:215]: This regex looks strange to me, what does it test?

Comment (niclastegner) [reply lib/manager/gradle/build-gradle.ts:215]: Sometimes packages have strange versioning, e.g. this one:
`org.springframework.cloud:spring-cloud-dependencies:Hoxton.SR8`

Will your solution handle that?

Comment (rarkins) [reply lib/manager/gradle/build-gradle.ts:215]: To what extend can we leverage our existing gradle versioning to determine if a version is valid?

Comment (zharinov) [reply lib/manager/gradle/build-gradle.ts:215]: Variable name can be valid version too. I'm planning to change #6477 implementation a little bit, such that we don't need to care about this.

**Comment by zharinov** [comment]
Unfortunately can't test the branch until weekend, the traffic is pretty heavy for my limited bandwidth 🙃

File: lib/manager/gradle/build-gradle.ts (hunk 1 above)
Comment (rarkins) [reply lib/manager/gradle/build-gradle.ts]: @zharinov should that change be done first?

Comment (zharinov) [reply lib/manager/gradle/build-gradle.ts]: Do you mean separate PR? It will break existing extractions, so pretty much PRs will be closed

**Comment by zharinov** [comment]
Tested for docker binary source too: https://github.com/zharinov/example-gradle-with-renovate/pull/1

File: lib/manager/gradle/build-gradle.ts [hunk 2]
```diff
@@ -167,7 +168,18 @@ function dependencyStringVariableExpressionFormatMatch(
   dependency: GradleDependency
 ): RegExp {
   return regEx(
-    `\\s*dependency\\s+['"]${dependency.group}:${dependency.name}:([^'"]+)['"](?:\\s|;|})`
+    `\\s*dependency\\s+['"]${dependency.group}:${dependency.name}:` +
+      // eslint-disable-next-line no-template-curly-in-string
+      '${([^}]*)}' +
```
Comment (zharinov) [inline lib/manager/gradle/build-gradle.ts:173]: `com.example:package:${varName}`

File: lib/manager/gradle/build-gradle.ts [hunk 3]
```diff
@@ -211,8 +223,14 @@ export function collectVersionVariables(
...
     }
 
-    if (!dep.currentValue && variables[depName]) {
-      dep.currentValue = variables[depName];
+    if (!dep.currentValue) {
+      const dependencyLiteralRegex = dependencyStringLiteralExpressionFormatMatch(
+        dependency
+      );
+      const currentValue = dependencyLiteralRegex.exec(buildGradleContent)?.[1];
+      if (gradleVersioning.isValid(currentValue)) {
```
Comment (rarkins) [inline lib/manager/gradle/build-gradle.ts:231]: We normally delay `isValid` checks until the process/lookup stage. Is it necessary to filter early here?

**Review by rarkins — CHANGES_REQUESTED** [review]
How risky is this change for backwards compatibility?

File: lib/manager/gradle/build-gradle.ts (hunk 3 above)
Comment (zharinov) [reply lib/manager/gradle/build-gradle.ts:231]: Should not be a problem if we get rid of it here

**Comment by zharinov** [comment]
> How risky is this change for backwards compatibility?

Should work as usual and additionally should support variables defined like `${thisOne}`

**Comment by rarkins** [comment]
@zharinov does this still lgtm?

**Review by viceice — APPROVED** [review]

**Pull request merged**